native/cocos/platform/android/java/vendor/google/billing/GoogleBillingHelper.java

```diff
@@ -24,191 +24,512 @@ of this software and associated documentation files (the "Software"), to deal
-    public static native void onBillingConfigResponse(@NonNull BillingResult billingResult, @Nullable BillingConfig billingConfig);
-
-    public static native void onAlternativeBillingOnlyTokenResponse(@NonNull BillingResult billingResult,
+
+    private static SparseArray<GoogleBilling> googleBillings = new SparseArray<GoogleBilling>();
+    private static int billingTag = 0;
+
+    public static int newTag() {
+        return billingTag++;
+    }
+    public static void createBillingClient(int tag, BillingClient.Builder builder) {
+        GlobalObject.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                GoogleBilling billing = new GoogleBilling(tag, builder);
+                googleBillings.put(tag, billing);
+            }
+        });
+    }
+
+    public static int removeBillingClient(int tag) {
+        final int index = billingTag;
+        GlobalObject.runOnUiThread(new Runnable() {
+            @Override
+            public void run() {
+                googleBillings.remove(tag);
+            }
+        });
+        return billingTag++;
+    }
+
+    public static final class BillingClientPurchasesUpdatedListener implements PurchasesUpdatedListener {
+        private int _tag;
+        BillingClientPurchasesUpdatedListener(int tag) {
+            this._tag = tag;
+        }
+        @Override
+        public void onPurchasesUpdated(@NonNull BillingResult billingResult, @Nullable List<Purchase> purchaseList) {
+            int tag = this._tag;
+            CocosHelper.runOnGameThread(new Runnable() {
+                @Override
+                public void run() {
+                    GoogleBilling billing = googleBillings.get(tag);
+                    if (billing != null) {
+                        int startID = billing.pushPurchases(purchaseList);
+                        GoogleBillingHelper.onPurchasesUpdated(tag, billingResult, purchaseList, startID);
+                    }
+                }
+            });
+        }
+    }
+
+    public static final class BillingClientUserChoiceBillingListener implements UserChoiceBillingListener {
+        private int _tag;
+        BillingClientUserChoiceBillingListener(int tag) {
+            this._tag = tag;
+        }
+        @Override
+        public void userSelectedAlternativeBilling(@NonNull UserChoiceDetails userChoiceDetails) {
+            int tag = this._tag;
+            CocosHelper.runOnGameThread(new Runnable() {
+                @Override
+                public void run() {
+                    GoogleBilling billing = googleBillings.get(tag);
+                    if (billing != null) {
+                        GoogleBillingHelper.userSelectedAlternativeBilling(tag, userChoiceDetails);
+                    }
+                }
+            });
+        }
+    }
+
+    public static native void onBillingSetupFinished(int tag, int callbackId, @NonNull BillingResult billingResult);
```

Comment:
Define it as private for internal use. Additionally, it's recommended to append "native" to the interface name to distinguish it from other interfaces with the same name.

Reply:
fixed.